syft/pkg/cataloger/golang/scaffolding_parser.go

```diff
@@ -0,0 +1,216 @@
+package golang
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/anchore/syft/internal/log"
+	"github.com/anchore/syft/syft/artifact"
+	"github.com/anchore/syft/syft/pkg"
+	"github.com/anchore/syft/syft/pkg/cataloger/common/command"
+	"github.com/anchore/syft/syft/source"
+)
+
+const (
+	GoSumFileName = "go.sum"
+	GoCommand     = "go"
+)
+
+func parseScaffolding(location source.Location, reader io.Reader) ([]*pkg.Package, []artifact.Relationship, error) {
+	currentFilepath := string(location.VisibleRealPath())
+	parser := scaffoldingParser{
+		currentFilepath: currentFilepath,
+		currentFileDir:  filepath.Dir(currentFilepath),
+		reader:          reader,
+		command:         GoCommand,
+	}
+	return parser.parse()
+}
+
+type scaffoldingParser struct {
+	currentFilepath string
+	currentFileDir  string
+	reader          io.Reader
+	command         string
+}
+
+func (p *scaffoldingParser) parse() ([]*pkg.Package, []artifact.Relationship, error) {
+	// lock file isn't existed
+	if _, err := os.Stat(filepath.Join(filepath.Dir(p.currentFilepath), GoSumFileName)); err != nil {
+		return parseGoMod(p.currentFilepath, p.reader)
+	}
+
+	log.Infof("parsing golang dependency file %s by scaffolding", p.currentFilepath)
+	var ctx, cancel = context.WithTimeout(context.Background(), 5*time.Minute)
+	defer cancel()
+
+	log.Info("parsing golang root module for %s", p.currentFilepath)
+	mainPkg, err := p.parseRootModule(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	log.Info("parsing golang all modules for %s", p.currentFilepath)
+	moduleMap, err := p.parseAllModules(ctx)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	log.Info("generating and parsing golang dependency tree for %s", p.currentFilepath)
+	pkgs, relationships, err := p.parseModuleRelationships(ctx, mainPkg, moduleMap)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	log.Infof("finished parsing golang dependency file %s, found %d packages, %d relationships", p.currentFilepath, len(pkgs), len(relationships))
+	return pkgs, relationships, nil
+}
+
+func (p *scaffoldingParser) parseRootModule(ctx context.Context) (*pkg.Package, error) {
+	buf := new(bytes.Buffer)
+	defer buf.Reset()
+
+	cmd, err := p.getRootModCommand(ctx, buf)
+	if err != nil {
+		return nil, err
+	}
+
+	if err = cmd.Run(); err != nil {
```

Comment:
Should we handle stderr? This is usually the ExitError that does not provide sufficient info

Reply:
I mean cmd.Stderr. Printing the command does not help gathering the error message